src/main/java/core/basesyntax/filereader/FileReaderImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.filereader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReaderService {
+
+    @Override
+    public List<String> read(String path) {
+        if (path.isEmpty() || path == null) {
+            throw new RuntimeException("Path doesn't exist");
+        }
+
+        List<String> stringLines = new ArrayList<>();
+
+        try {
+            stringLines = Files.readAllLines(Path.of(path));

```

The code does not check if the file actually exists and is readable before attempting to read. Requirement 3.2 specifies that the method should validate the file path and throw a RuntimeException with an informative message if invalid.